syllabiShare/models.py

```diff
@@ -34,8 +40,7 @@ def upload(self,name):
         else:
             self.uploads[name] = 1
     def topFive(self):
-        return [i for i in sorted(self.uploads.items(), key=lambda x: x[1], reverse=True)[:min(len(self.uploads),5)]]
-        
+        return [i for i in sorted(self.uploads.items(), key=lambda x: x[1], reverse=True)[:min(len(self.uploads),5)] & self.settings.optOut != True ]
```

Comment:
The `&` should be `and`, right?

Also, not entirely following how this opts out specific people, it seems to be a school-wide opt-out instead?

Reply:
I don't really understand what this line does, as I was about to test it and got an error. What is a way that I could make it catered to a specific person?

Reply:
This got me on a pretty large tangent trying to figure out a more Django-y way to do it. I think the best way to do it would be through a Django `filter` on a School's Users that aren't opted out, ordered by submission count.
...unfortunately, there is no User->School relation yet, so that was the rabbit hole that I dug/am digging for myself.

What's also needed though is a way to associate the opt-out setting to a specific user, so it feels to me like you'll need that `user` field on Settings to be a ForeignKey (or OneToOne? I think OneToOne but my brain is tired) instead.
(And if it's OneToOne, then should it be a `settings` field on User, or a `user` field on Settings?)

Reply:
Haha yeah, it seems convoluted. But that makes sense. I had a foreign key from my settings model to the school model, but I'm not sure of any other way to do it and Vernon doesn't seem like he is a fan of foreign keys.

Reply:
haha yeah I avoided them like the plague in 3240, but its probably needed here to scale